common/app/model/dotcomrendering/DotcomRenderingDataModel.scala

```diff
@@ -26,9 +26,12 @@ case class DotcomRenderingDataModel(
     editionLongForm: String,
     editionId: String,
     pageId: String,
+    // Format and previous flags
+    format: Map[String, String],
```

Comment:
Rather than a `Map[String, String]` could we use a custom `case class`? It's not wrong to use a map, just unidioamatic in Scala as people tend to prefer the type safety of an actual class.

Reply:
This sounds like a much better idea. I'd been thinking along these lines originally, but because everything else in the data model were `String` based, I assumed I should do the same. I'll take your advice and implement a `writes` method and use that. Thanks!